### PR TITLE
AV-202495 check object type before the typecast

### PR DESCRIPTION
**Pull request by arihantg**
Manually tested with a expired certificate in secure ingress .

### CODE REVIEW

**Comment by arihantg** [comment]
build ako

**Comment by arihantg** [comment]
build ako

File: internal/rest/dequeue_nodes.go [hunk 1]
```diff
@@ -699,7 +699,15 @@ func (rest *RestOperations) ExecuteRestAndPopulateCache(rest_ops []*utils.RestOp
 
 func updateGatewayStatusWithVsError(key string, rest_op *utils.RestOp) {
 	if rest_op.Model == "VirtualService" {
-		vs := rest_op.Obj.(avimodels.VirtualService)
+		var vs avimodels.VirtualService
+		switch rest_op.Obj.(type) {
+		case avimodels.VirtualService:
+			vs = rest_op.Obj.(avimodels.VirtualService)
```
Comment (pkoshtavmware) [inline internal/rest/dequeue_nodes.go:705]: Please describe in which case object type will be *avimodels.VirtualService and avimodels.VirtualService ?

**Comment by pkoshtavmware** [comment]
> Manually tested with a expired certificate in secure ingress .

Isn't this function used for gateway only ?

File: internal/rest/dequeue_nodes.go (hunk 1 above)
Comment (arihantg) [reply internal/rest/dequeue_nodes.go:705]: in case of child vs it will be *avimodels.VirtualService while in parent vs it is avimodels.VirtualService

**Comment by arihantg** [comment]
> > Manually tested with a expired certificate in secure ingress .
> 
> Isn't this function used for gateway only ?

It is the case after the merge of https://github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pull/1390. I just tested without pulling this changeset.

**Review by pkoshtavmware — APPROVED** [review]
LGTM

**Review by DixitAakash — APPROVED** [review]
LGTM

File: internal/rest/dequeue_nodes.go [hunk 2]
```diff
@@ -699,7 +699,15 @@ func (rest *RestOperations) ExecuteRestAndPopulateCache(rest_ops []*utils.RestOp
...
+		case *avimodels.VirtualService:
```
Comment (akshayhavile) [inline internal/rest/dequeue_nodes.go:706]: As logic at line number 711 is for Parent VS only, should we return from line number 707? no need to check type for child vs.. right?

Comment (arihantg) [reply internal/rest/dequeue_nodes.go:706]: let's keep it as defensive check in case there is unexplored scenario for parentvs  where type can be pointer.

**Review by pkoshtavmware — APPROVED** [review]

**Pull request merged**